syntax = "proto3";
package message;

/*prepared for verification service*/
service VerificationService {
  rpc GetVerificationCode(GetVerificationRequest)
      returns (GetVerificationResponse) {}
}

message GetVerificationRequest { string email = 1; }

message GetVerificationResponse {
  int32 error = 1;
  string email = 2;
  string message = 3;
}

service UserService {
  // Register new user UUID and get allocated chatting server
  rpc RegisterUser(UserRegisterRequest) returns (UserRegisterResponse);

  // User login
  rpc LoginUser(LoginRequest) returns (LoginResponse);
}

message UserRegisterRequest { int32 uuid = 1; }

message UserRegisterResponse {
  int32 error = 1;
  string host = 2;
  string port = 3;
  string token = 4;
}

message LoginRequest {
  int32 uuid = 1;
  string token = 2;
}

message LoginResponse { int32 error = 1; }

message LogoutRequest {
  int32 uuid = 1;
  string token = 2;
}

message LogoutResponse { int32 error = 1; }